Added spatial analysis tutorial notebook

tomp opened the pull request:
This is currently just Armaan's original notebook,  updated for the current state of luna.  It still uses the CLIs. 

The original notebook is `[spatial_stat_pdl1_example.ipynb](https://github.mskcc.org/gist/kohlia/7657bf52522e8f2a24523ba9098b0782)`

Commit:
Added spatial analysis tutorial notebook

This is just Armaan's original notebook, updated for the current
state of luna.  It still uses the CLIs.

Commit:
Rewrote spatial tutorial notebook as Python code, and removed UMAP calcs

1. Added function to pull output file names from metadata.yml files.
2. Cast Path objects to strings in a couple of places.
3. Check for both `aperio.MPP` and `openslide.mpp-x` in the slide
   image properties, and log a warning if neither is found.
4. Added new notebook to the tutorial docs
5. Renamed notebook to spatial_stats.ipynb

armaank commented on pyproject.toml:
Instead of installing umap-learn as a dependency here, it might be better to add `!pip3 install umap-learn` to the notebook, since it's a lib we don't use anywhere else but the tutorial.

armaank commented on src/luna/pathology/cli/visualize_tile_labels_png.py:
```diff
@@ -123,15 +125,17 @@ def visualize_tiles(
         # See if we need to adjust scale_factor to account for different units
         if mpp_units:
-            unit_sf = float(slide.properties["openslide.mpp-x"])
+            unit_sf = float(slide.properties["aperio.MPP"])
```
Instead of removing the old slide properties, can you add a check to see if `openslide.mpp-x` is in the properties, and if not use `aperio.MPP` - different libs for loading/saving slides have slightly different metadata so it'd be good to check for both

armaank commented on docs/tutorials/spatial_stats_pdl1.ipynb:
A few notes:
a) instead of using the `read_metadata()` function, it actually might be better to check the outputs via `ls`, since it shows users all of the outputs the tool
b) You can remove the original cli calls
c) We actually don't really need the UMAP section - I think the output image from the visualize tile labels illustrates the point well enough.

armaank commented:
One last item we need to address before we finalize this is the actual doc build. If you run `make docs` - then you should be able to see the tutorial on the doc site along-side the others.

armaank commented on docs/tutorials/.pages:
```diff
@@ -9,4 +9,5 @@ nav:
     - Snakemake: '5_end-to-end-pipeline.ipynb'
     - Digital Slide Archive (DSA) Visualization: '6_dsa-tools.ipynb'
     - DSA Annotation: '7_dsa-annotation.ipynb'
+    - Spatial Analysis: 'spatial_stats_pdl1.ipynb'
```
rename the notebook to just `spatial_stats.ipynb`

armaank reviewed (approved):
Looks good!

Commit:
Allow non-local connections to the docs server

This lets us run a docs server on one of the compute servers and view
the docs site on our laptops.
